src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java

```diff
@@ -203,6 +206,58 @@ public static void copyFile(String sourcePath, String destPath) throws IOExcepti
         }
     }
 
+    /**
+     * If a provided path makes reference to a remote resource, it copies it over to a temporary local file that
+     * is returned.
+     * <p>
+     *     The returned temporary file will be marked for deletion on exit if so requested.
+     * </p>
+     * <p>
+     *     If the source is actually a local file it is considered to be already staged, unless forceStaggingOfLocalFiles is true,
+     *     and it simply returns the corresponding File object, neither the deletion
+     *     on exit request nor the tempDir input have any effect.
+     * </p>
+     * <p>
+     *     In order to distinguish between these two scenarios you must call {@link BucketUtils#isCloudStorageUrl(String)}
+     *     independently.
+     * </p>
+     * @param sourcePath
+     * @param tempDir if provided the temporary file is created under this directory, if needed. If {@code null} the system
+     *                default temporary file location is used instead.
+     * @return never {@code null}.
+     * @throws IOException if such an exeception occurs.
+     */
+    public static File stageFile(String sourcePath, final File tempDir, final boolean markForDeletionOnExit, final boolean forceStaggingOfLocalFiles) throws IOException {
```

Comment:
I don't understand why we need this method.  In general I'm against adding anything to BucketUtils which is essentially deprecated code in my mind.  

This method has a ton of parameters but it's only used in 1 place with specific hardcoded options.  It's not tested at all.  

I would prefer using Files.copy(file, IOUtils.createTempPath(something)) since the only usage of this method always makes a copy and marks it for deletion.

It seems like where we're using this we might be able to use the ZipFileSystem instead of ZipFile anyway.  I'm not sure if it can read from a remote path natively but it might...

Reply:
Looks like we forgot to delete this, doing it now (its unused i think).